### ISSUE TITLE
[bug] MesonToolchain cannot set `-mmacosx-version-min` flag automatically when build for macos target

### ISSUE THREAD

**Issue by zhouxiasai** [conan-io/conan#15459]
### Environment details

* Operating System+version: macOS Sonoma 14.2.1
* Compiler+version: Xcode 15.1
* Conan version: 2.0.13
* Python version: 3.10.5

### Steps to reproduce

1. Build lib dav1d from conan-center-index
2. check conan_meson_cross.ini - No flag found
3. otool -l ${path_to_libdav1d.dylib} - minos is 14.0

### Logs

In file conan/tools/meson/toolchain.py 

line 304:
```
        os_sdk = self._conanfile.settings.get_safe('os.sdk')
        if not os_sdk and self._os != "Macos":
            raise ConanException("Please, specify a suitable value for os.sdk.")
```
When built for macos, there is no os.sdk in settings.yml, so this os_sdk is None

line 314
```
        self.apple_min_version_flag = [apple_min_version_flag(os_version, os_sdk, subsystem)]
```
In function `apple_min_version_flag`
```
def apple_min_version_flag(os_version, os_sdk, subsystem):
    """compiler flag name which controls deployment target"""
    if not os_version or not os_sdk:
        return ''

    # FIXME: This guess seems wrong, nothing has to be guessed, but explicit
    flag = ''
    if 'macosx' in os_sdk:
        flag = '-mmacosx-version-min'
    elif 'iphoneos' in os_sdk:
        flag = '-mios-version-min'
    elif 'iphonesimulator' in os_sdk:
        flag = '-mios-simulator-version-min'
    elif 'watchos' in os_sdk:
        flag = '-mwatchos-version-min'
    elif 'watchsimulator' in os_sdk:
        flag = '-mwatchos-simulator-version-min'
    elif 'appletvos' in os_sdk:
        flag = '-mtvos-version-min'
    elif 'appletvsimulator' in os_sdk:
        flag = '-mtvos-simulator-version-min'
    elif 'xros' in os_sdk:
        # need early return as string did not fit into the "normal" schema
        return f'-target arm64-apple-xros{os_version}'
    elif 'xrsimulator' in os_sdk:
        # need early return as string did not fit into the "normal" schema
        return f'-target arm64-apple-xros{os_version}-simulator'

    if subsystem == 'catalyst':
        # especial case, despite Catalyst is macOS, it requires an iOS version argument
        flag = '-mios-version-min'

    return f"{flag}={os_version}" if flag else ''
```
If os_sdk is none, it returns ''
while expected `-mmacosx-version-min={os_version}`

So, MesonToolchain never write -mmacosx-version-min={os_version}  to the meson_build.ini, which will be compiler default (eg. 14.0)

**Comment by franramirez688** [comment]
Hi @zhouxiasai 

I'd need more information to figure out the problem:

* Could you please provide both Conan profiles host and build used for this library?
* The command you're running, e.g., `conan install --requires dav1d/1.3.0 -pr:h host -pr:b build`.
* Are you running into errors or are you simply not finding the `-mmacosx-version-min` in the `conan_meson_cross.ini`file?

Any other information to understand the issue better is more than welcome.

**Comment by zhouxiasai** [comment]
Hi @franramirez688 
Thanks for reply.

Here is my settings.yml [settings.yml.txt](https://github.com/conan-io/conan/files/13947776/settings.yml.txt)
My profiles are:
```
======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu14
compiler.libcxx=libc++
compiler.version=15
os=Macos
os.version=10.11

Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=15
os=Macos
```

My command is `conan build conanfile_v2.py -pr macos -s arch=x86_64`. 
The profile macos is here [macos.txt](https://github.com/conan-io/conan/files/13947813/macos.txt)
The conanfile_v2.py is here [conanfile_v2.py.txt](https://github.com/conan-io/conan/files/13947852/conanfile_v2.py.txt)

No error occurs, but when I check it with `otool -l ${path_to_the_binary}` and try to find `LC_VERSION_MIN_MACOSX` for x86_64 and `minos` for arm64, it shows 14.0 (which is default value by Xcode 15).

So I manually set `tc.apple_min_version_flag = [f"-mmacosx-version-min={self.settings.os.version}"]` to change this value to my destination os.version 10.11. It should be done in the toolchain by default.

**Comment by franramirez688** [comment]
Hi @zhouxiasai 

Thanks a lot for all the details 👏 

Yes, it was a bug. That flag should exist when cross-compiling between different architectures in macOS. Thanks for reporting.

I just opened a PR https://github.com/conan-io/conan/pull/15465 that would solve this issue. It'll be released in the next Conan 2.1 version.
